nms/app/views/alarms/components/hooks.ts

```diff
@@ -77,6 +77,57 @@ export function useLoadRules<TRuleUnion>({
   };
 }
 
+/**
+ * Filter a route from routes for a especific ruleName and receiver
+ * @param {*} response from apiUtil.useAlarmsApi
+ * @param {*} initialReceiver string
+ * @param {*} ruleName string
+ * @returns
+ */
+export function filterRouteByRuleName(
```

Comment:
Hi Thanks for feedback @thmsschmitt . Could you help on this? My _Eslint_ is complaining about the funciton _filterUpdatedFilterRoutes_ because on ' //update existing route' blocks the variables **initialReceiver** and **receiver** can be null or undefined. So because of that I made all these verification. But analyzing what you said, is it enough if i just use the **initialReceiver!** and **receiver!** to prevent null and undefined while calls the _filterUpdatedFilterRoutes_?


```
// update existing route
      const _updatedRoutes = filterUpdatedFilterRoutes(
        response,
        initialReceiver!,
        ruleName,
        receiver!,
      );
      updatedRoutes = {
        ...updatedRoutes,
        routes: _updatedRoutes,
      };
```

Reply:
No not really, there are three branches:

-  receiver === null && initialReceiver !== null
-  receiver !== null &&  initialReceiver === null 
-  receiver === null && initialReceiver === null || receiver !== null && initialReceiver !== null

`filterUpdatedFilterRoutes` is called in the last case so it needs to handle the case of both being null.
`filterRouteByRuleName` is called in the first case so it can assume that initialReceiver is not null.

`undefined` should be taken out of the picture by setting up the states like so:

```
const [initialReceiver, setInitialReceiver] = React.useState<string | null>(null);
const [receiver, setReceiver] = React.useState<string | null>(null); 
```